grpclog/loggerv2_test.go

```diff
@@ -61,3 +66,264 @@ func checkLogForSeverity(s int, b []byte) error {
 	}
 	return nil
 }
+
+// makeLogFuncErrr returns a logger printFunc which generates a test error,
+// for testing that the log func is not called
+func makeLogFuncErr(t *testing.T) func() string {
+	return func() string {
+		t.Errorf("want Logger func to not be called for io.Discard")
+		return ""
+	}
+}
+
+// logFuncStr is a string used via testLogFuncOk and
+// testCheckLogContainsFuncStr to test the logger printFunc.
+const logFuncStr = "called-func"
+
+// logFuncOk is a logger printFunc to write a constant string
+func logFuncOk() string {
+	return logFuncStr
+}
+
+// checkLogContainsFuncStr checks that the logger buffer logBuf contains the
+// string written by logFuncOk
+func checkLogContainsFuncStr(t *testing.T, logBuf []byte) {
+	if !bytes.Contains(logBuf, []byte(logFuncStr)) {
+		t.Errorf("want logger func to be called and print '%v', got '%v'", logFuncStr, string(logBuf))
+	}
+}
+
+// checkWrite checks that the log buffer buf was written as expected.
```

Comment:
nit: the docstring is too verbose. My suggestion would be to be more explicit in naming the function and write inline comment where you are calling the function stating what you are trying to compare

Reply:
Changed